src/wildcard.cpp

```diff
@@ -139,9 +156,118 @@ bool wildcard_match(const wcstring &str, const wcstring &wc, bool leading_dots_f
                 if (is_first && *str_x == L'.') {
                     return false;
                 }
-                wc_x++;
-                str_x++;
-                continue;
+                if (!no_slashes_wildcard || *str_x != L'/') {
+                    wc_x++;
+                    str_x++;
+                    continue;
+                }
+            } else if (extended && *wc_x == WILDCARD_RANGE_START && *str_x != 0) {
```

Comment:
Pretty sure those should both match, and I think it runs awfoul of the awkward leading dots rule.

(there's potential for divering from posix globs, specifically in matching leading dots like that and leading slashes with ranges - `[/]foo` is supposed *not* to match `/foo`, which just seems like an awkward and unnecessary limitation - *maybe* for ranges and classes it makes sense, but if there's a literal `/` I can't see a possible justification)

Reply:
Ah, it's like the comment right below says:

```c++
                // Can't match a "/" or a leading ".", or even a "." after a "/",
                // *even if* it's literally in the class!
```

So the first is actually buggy, and all of this is ridiculous.

Reply:
Simple off-by-one:

```c++
str_x - str.c_str() > 1 && *str_x == L'.' && *(str_x - 1) == L'/'
```

`str.c_str()` is already valid, so to figure out if we have one valid char before (to avoid going too far back) we need `str_x - str.c_str() > 0`.